src/test/java/uk/gov/justice/digital/delius/service/ConvictionService_GetProbationStatus.java

```diff
@@ -0,0 +1,219 @@
+package uk.gov.justice.digital.delius.service;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.justice.digital.delius.config.FeatureSwitches;
+import uk.gov.justice.digital.delius.data.api.ProbationStatus;
+import uk.gov.justice.digital.delius.entitybuilders.EventEntityBuilder;
+import uk.gov.justice.digital.delius.entitybuilders.KeyDateEntityBuilder;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Disposal;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Event;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Offender;
+import uk.gov.justice.digital.delius.jpa.standard.repository.EventRepository;
+import uk.gov.justice.digital.delius.jpa.standard.repository.OffenderRepository;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class ConvictionService_GetProbationStatus {
+
+    private static final String CRN = "CRN";
+
+    private ConvictionService convictionService;
+
+    @Mock
+    private EventRepository eventRepository;
+
+    @Mock
+    private OffenderRepository offenderRepository;
+
+    @Mock
+    private EventEntityBuilder eventEntityBuilder;
+
+    @Mock
+    private SpgNotificationService spgNotificationService;
+
+    @Mock
+    private LookupSupplier lookupSupplier;
+
+    @Mock
+    private IAPSNotificationService iapsNotificationService;
+
+    @Mock
+    private ContactService contactService;
+
+    @Mock
+    private TelemetryClient telemetryClient;
+
+    @Mock
+    private Offender offender;
+
+    @Mock
+    private Event event;
+    @Mock
+    private Event event2;
+    @Mock
+    private Event event3;
+    @Mock
+    private Disposal disposal;
+
+    @BeforeEach
+    public void setUp(){
+        final var featureSwitches = new FeatureSwitches();
+        convictionService = new ConvictionService(eventRepository, offenderRepository, eventEntityBuilder, spgNotificationService, lookupSupplier, new KeyDateEntityBuilder(lookupSupplier), iapsNotificationService, contactService, telemetryClient, featureSwitches);
+    }
+
+    @Test
+    public void canGetProbationStatusForCurrentOffenderInBreach() {
+        when(offenderRepository.findByCrn(CRN)).thenReturn(Optional.of(offender));
```

Comment:
Decided to go with mocks rather than real objects for this test to make it explicit which method calls were relevant as there's a reasonably amount of logic. Also strict mocking means I know there's no redundant calls (this can be a bit temperamental but I tried adding unused mocks and it always picked it up so seems to be working as intended...). In retrospect I could dispense with the comments and simplify these tests by extracting the bulk of the logic to a helper class to improve readability if needed.

Reply:
Whatever you feel comfortable with.